Bottom: a8043384706cbb2a03c388849e3b8eee4a7e79da
Top:    3cbb32e77456133e5cc8da40ab9d72795e588f07
Author: Ian Kent <raven@themaw.net>
Date:   2013-02-27 12:58:49 +0800

Refresh of autofs-5.0.7-fix-wildcard-multi-map-regression.patch

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index ef64bf1..46771b6 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1102,7 +1102,7 @@ do_cache_lookup:
 		 * Add wildcard match for later validation checks and
 		 * negative cache lookups.
 		 */
-		if (ap->type == LKP_INDIRECT && me->key == '*') {
+		if (ap->type == LKP_INDIRECT && *me->key == '*') {
 			ret = cache_update(mc, source, key, me->mapent, me->age);
 			if (!(ret & (CHE_OK | CHE_UPDATED)))
 				me == NULL;
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index c69b9f8..d7c2cf6 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -2984,7 +2984,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		 * Add wildcard match for later validation checks and
 		 * negative cache lookups.
 		 */
-		if (ap->type == LKP_INDIRECT && me->key == '*') {
+		if (ap->type == LKP_INDIRECT && *me->key == '*') {
 			ret = cache_update(mc, source, key, me->mapent, me->age);
 			if (!(ret & (CHE_OK | CHE_UPDATED)))
 				me == NULL;
diff --git a/modules/lookup_nisplus.c b/modules/lookup_nisplus.c
index 457bb0b..735a1b3 100644
--- a/modules/lookup_nisplus.c
+++ b/modules/lookup_nisplus.c
@@ -576,7 +576,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		 * Add wildcard match for later validation checks and
 		 * negative cache lookups.
 		 */
-		if (ap->type == LKP_INDIRECT && me->key == '*') {
+		if (ap->type == LKP_INDIRECT && *me->key == '*') {
 			ret = cache_update(mc, source, key, me->mapent, me->age);
 			if (!(ret & (CHE_OK | CHE_UPDATED)))
 				me == NULL;
diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 77e9a93..cca862a 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -650,7 +650,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		 * Add wildcard match for later validation checks and
 		 * negative cache lookups.
 		 */
-		if (ap->type == LKP_INDIRECT && me->key == '*') {
+		if (ap->type == LKP_INDIRECT && *me->key == '*') {
 			ret = cache_update(mc, source, key, me->mapent, me->age);
 			if (!(ret & (CHE_OK | CHE_UPDATED)))
 				me == NULL;
